### PR TITLE
Add Cavat review

### PR DESCRIPTION
**Pull request by ThierryJudge**

### CODE REVIEW

File: article/_posts/2022-01-14-CAVAT.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+et al., 2018 (Prior-aware segmentation) and Painchaud et al. 2020. 
+
+Authors explore the aspect of semi-supervised learning as training segmentation networks to learn valid shapes is a 
+difficult task when little labeled data is available. Authors, therefore, propose a method base on Virtual Adversarial 
+Training (VAT) and REINFORCE to optimize a non-differentiable local connectivity constraint. 
+
+# Method 
+
+## Note on VAT 
+The goal of the VAT paper (Virtual Adversarial Training: A Regularization Method for Supervised and Semi-Supervised
```
Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:30]: If the paper is so relevant, adding the reference would be useful I think.

File: article/_posts/2022-01-14-CAVAT.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+Training the proposed method is done on a labeled set $$\matcal{S}$$ and an un-labled set $$mathcal{U}$$. The network is 
+trained using the following loss function.
+
+![](/article/images/cavat/equation1.JPG)
+
+The supervised loss $$\mathcal{L}_{sup}$$. The Context-aware VAT loss term is composed of two losses. 
+
+![](/article/images/cavat/equation3.JPG)
+
+The first loss is the local distribution smoothness (LSD), which is directly taken from VAT. This loss regularized the
```
Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:51]: regularized -> regularizes

File: article/_posts/2022-01-14-CAVAT.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+where $$J$$ is the per-pixel reward and $$p_u = f(x_u+r_r;\theta)$$ is the probability output of the network. 
+
+
+Looking back at equation (3), we can note here that the adversarial perturbation $$r_U$$ is maximized on both terms of 
+the CAVAT loss which increases the number of samples that violate the constraint. 
+
+## Local connectivity constraint
+
+The constraint optimized in this paper is the local connectivity constraint. Connectivity if a region G means that there
```
Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:68]: if -> of ?

File: article/_posts/2022-01-14-CAVAT.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+Looking back at equation (3), we can note here that the adversarial perturbation $$r_U$$ is maximized on both terms of 
+the CAVAT loss which increases the number of samples that violate the constraint. 
+
+## Local connectivity constraint
+
+The constraint optimized in this paper is the local connectivity constraint. Connectivity if a region G means that there 
+exists a path between every pair of pixels in G such that every pixel in that path is also in G. As global connectivity
+is hard to achieve at the beginning of training, authors propose to optimize local connectivity at local patches.
+
+Given a discreate prediction $$\hat{y}$$. The others apply a $$1_{ l\times l}$$ convolution to find the pixel with the
```
Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:72]: descreate -> discrete

Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:72]: given a discreate prediction $$\hat{y}$$. -> Unfinished sentence.

Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:72]: The others apply a (...) -> "The others" does not read well as a subject: They? Other authors?

File: article/_posts/2022-01-14-CAVAT.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+## Local connectivity constraint
+
+The constraint optimized in this paper is the local connectivity constraint. Connectivity if a region G means that there 
+exists a path between every pair of pixels in G such that every pixel in that path is also in G. As global connectivity
+is hard to achieve at the beginning of training, authors propose to optimize local connectivity at local patches.
+
+Given a discreate prediction $$\hat{y}$$. The others apply a $$1_{ l\times l}$$ convolution to find the pixel with the 
+most neighbors ($$l=5$$). They then apply a flood-filling algotithm starting at this pixel to identify its region C. For 
+each $$k \times K$$ patch, the authors find the pixels (S) not in C using a convolution
```
Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:74]: Is the second `K` meant to be in capitals?

File: article/_posts/2022-01-14-CAVAT.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+exists a path between every pair of pixels in G such that every pixel in that path is also in G. As global connectivity
+is hard to achieve at the beginning of training, authors propose to optimize local connectivity at local patches.
+
+Given a discreate prediction $$\hat{y}$$. The others apply a $$1_{ l\times l}$$ convolution to find the pixel with the 
+most neighbors ($$l=5$$). They then apply a flood-filling algotithm starting at this pixel to identify its region C. For 
+each $$k \times K$$ patch, the authors find the pixels (S) not in C using a convolution 
+$$S=1_{k \times k} \circledast (\hat{y} - C)$$ ($$k=3$$). Finally the reward for each pixel $$i$$ is given by
+$$J_i(\hat{y}) = \delta(S_i) = 0) $$ where $$\delta(\cdot)$$ is the Kronecker delta. 
+# Results 
+Authors test their method on binary segmentation of the structures in the ACDC heart segmentation dataset (LV, RV, MYO)
```
Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:78]: segmentation -> segmentations

File: article/_posts/2022-01-14-CAVAT.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+Given a discreate prediction $$\hat{y}$$. The others apply a $$1_{ l\times l}$$ convolution to find the pixel with the 
+most neighbors ($$l=5$$). They then apply a flood-filling algotithm starting at this pixel to identify its region C. For 
+each $$k \times K$$ patch, the authors find the pixels (S) not in C using a convolution 
+$$S=1_{k \times k} \circledast (\hat{y} - C)$$ ($$k=3$$). Finally the reward for each pixel $$i$$ is given by
+$$J_i(\hat{y}) = \delta(S_i) = 0) $$ where $$\delta(\cdot)$$ is the Kronecker delta. 
+# Results 
+Authors test their method on binary segmentation of the structures in the ACDC heart segmentation dataset (LV, RV, MYO) 
+and on the Prostate MR Image Segmentation (PROMISE12) Challenge dataset. 
+
+They compare and combine their method with other state of the art semi-supervised learning and evaluation on Dice score
```
Comment (jhlegarreta) [inline article/_posts/2022-01-14-CAVAT.md:81]: They compare and combine their method with other state of the art semi-supervised learning and evaluation (...) -> They compare and combine their method with other state of the art semi-supervised learning methods and evaluate (...) evaluation

Not sure if combine was meant to be there, though. If they combine their method with other methods, such an explanation should appear in the `Methods` section. At least a mention to it.

File: article/_posts/2022-01-14-CAVAT.md (hunk 1 above)
Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Fixed in bdcc473

File: article/_posts/2022-01-14-CAVAT.md (hunk 2 above)
Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Fixed in bdcc473

File: article/_posts/2022-01-14-CAVAT.md (hunk 3 above)
Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Fixed in bdcc473

File: article/_posts/2022-01-14-CAVAT.md (hunk 4 above)
Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Fixed in bdcc473

Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Fixed in bdcc473

Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Fixed in bdcc473

File: article/_posts/2022-01-14-CAVAT.md (hunk 5 above)
Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: No, Fixed in bdcc473

File: article/_posts/2022-01-14-CAVAT.md (hunk 6 above)
Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Fixed in bdcc473

File: article/_posts/2022-01-14-CAVAT.md (hunk 7 above)
Comment (ThierryJudge) [reply article/_posts/2022-01-14-CAVAT.md]: Should be combined but reworded in bdcc473

Comment (jhlegarreta) [reply article/_posts/2022-01-14-CAVAT.md]: OK, now the sentence/context makes sense/is clear. Thanks.

**Review by jhlegarreta — APPROVED** [review]

**Review by jodoin — APPROVED** [review]

**Pull request merged**